src/pages/Checkout/CheckoutBetaPage.vue

```diff
@@ -181,24 +180,29 @@ export default {
 	},
 	apollo: {
 		query: initializeCheckout,
-		preFetch: true,
-		result({ data, loading }) {
-			if (loading) {
-				this.loading = true;
-			}
+		// using the prefetch function form allows us to act on data before the page loads
+		preFetch(config, client) {
+			return client.query({
+				query: initializeCheckout
+			}).then(({ data }) => {
+				const totals = _get(data, 'shop.basket.totals');
+				// check for bonus credit and redirect if present
+				// TODO: remove once bonus credit functionality is added
+				// TODO: bonusAvailableTotal is reporting 0 once the credit has been removed in legacy basket
+				if (parseFloat(totals.bonusAvailableTotal) > 0) {
+					return Promise.reject({ path: '/basket?kexpn=checkout_beta.minimal_checkout&kexpv=a', query: {} });
```

Comment:
The parameters in the path can be moved to the query if you want:
`{ path: '/basket', query: { kexpn: 'checkout_beta.minimal_checkout', kexpv: 'a' } }`
If not you can also drop the empty `query`. This just needs to be a valid [VueRouter Route object](https://router.vuejs.org/api/#route-object-properties).

Reply:
good call...updating now